pkg/volume/azure_file/azure_file.go

```diff
@@ -139,6 +142,45 @@ func (plugin *azureFilePlugin) newUnmounterInternal(volName string, podUID types
 	}}, nil
 }
 
+func (plugin *azureFilePlugin) RequiresFSResize() bool {
+	return false
+}
+
+func (plugin *azureFilePlugin) ExpandVolumeDevice(
+	spec *volume.Spec,
+	newSize resource.Quantity,
+	oldSize resource.Quantity) (resource.Quantity, error) {
+
+	if spec.PersistentVolume != nil && spec.PersistentVolume.Spec.AzureFile == nil {
+		return oldSize, fmt.Errorf("invalid PV spec")
+	}
+	shareName := spec.PersistentVolume.Spec.AzureFile.ShareName
+	azure, err := getAzureCloudProvider(plugin.host.GetCloudProvider())
+	if err != nil {
+		glog.V(2).Infof("failed to get azure provider")
+		return oldSize, err
+	}
+
+	secretName, secretNamespace, err := getSecretNameAndNamespace(spec, spec.PersistentVolume.Spec.ClaimRef.Namespace)
+	if err != nil {
+		glog.V(2).Infof("failed to get secret")
+		return oldSize, err
+	}
+
+	util := &azureSvc{}
+	accountName, accountKey, err := util.GetAzureCredentials(plugin.host, secretNamespace, secretName)
+	if err != nil {
+		glog.V(2).Infof("failed to get account key for secret: %s in namespace: %s", secretName, secretNamespace)
+		return oldSize, err
+	}
+
+	if err := azure.ResizeFileShare(accountName, accountKey, shareName, int(newSize.Value()/1024/1024/1024)); err != nil {
```

Comment:
Can you use - https://github.com/kubernetes/kubernetes/blob/master/pkg/volume/util.go#L303 rather than dividing by 1024**3 ?

Reply:
fixed